cmd/cql-mysql-adapter/cursor.go

```diff
@@ -0,0 +1,311 @@
+	typeStr = strings.ToUpper(typeStr)
+
+	if strings.Contains(typeStr, "INT") {
+		return my.MYSQL_TYPE_LONGLONG
+	} else if strings.Contains(typeStr, "CHAR") || strings.Contains(typeStr, "CLOB") ||
+		strings.Contains(typeStr, "TEXT") {
+		return my.MYSQL_TYPE_VAR_STRING
+	} else if strings.Contains(typeStr, "BLOB") || typeStr == "" {
+		return my.MYSQL_TYPE_LONG_BLOB
+	} else if strings.Contains(typeStr, "REAL") || strings.Contains(typeStr, "FLOA") ||
+		strings.Contains(typeStr, "DOUB") {
+		return my.MYSQL_TYPE_DOUBLE
+	} else if strings.Contains(typeStr, "BOOLEAN") {
+		return my.MYSQL_TYPE_BIT
+	} else if strings.Contains(typeStr, "TIMESTAMP") || strings.Contains(typeStr, "DATETIME") {
+		return my.MYSQL_TYPE_TIMESTAMP
+	} else if strings.Contains(typeStr, "TIME") {
+		return my.MYSQL_TYPE_TIME
+	} else if strings.Contains(typeStr, "DATE") {
+		return my.MYSQL_TYPE_DATE
+	} else {
+		return my.MYSQL_TYPE_LONG_BLOB
+	}
+}
+
+// UseDB handle COM_INIT_DB command, you can check whether the dbName is valid, or other.
+func (c *Cursor) UseDB(dbName string) (err error) {
+	c.curDBLock.Lock()
+	defer c.curDBLock.Unlock()
+
+	// test if the database name is a valid database id
+	if !dbIDRegex.MatchString(dbName) {
+		// invalid database
+		return my.NewError(my.ER_BAD_DB_ERROR, fmt.Sprintf("invalid database: %v", dbName))
+	}
+
+	// connect database
+	cfg := client.NewConfig()
+	cfg.DatabaseID = dbName
+
+	var db *sql.DB
+
+	if db, err = sql.Open("covenantsql", cfg.FormatDSN()); err != nil {
+		return
+	}
+
+	c.curDB = dbName
+	c.curDBInstance = db
+
+	return
+}
+
+// HandleQuery handle COM_QUERY comamnd, like SELECT, INSERT, UPDATE, etc...
+// if Result has a Resultset (SELECT, SHOW, etc...), we will send this as the repsonse, otherwise, we will send Result.
+func (c *Cursor) HandleQuery(query string) (r *my.Result, err error) {
+	var conn *sql.DB
+
+	if conn, err = c.ensureDatabase(); err != nil {
+		return
+	}
+
+	// send empty result for variables query/table listing
+	if emptyResultQuery.MatchString(query) {
+		// return empty result
+		return &my.Result{
+			Status:       0,
+			InsertId:     0,
+			AffectedRows: 0,
+			Resultset:    nil,
+		}, nil
+	}
+
+	// use database query, same logic as COM_INIT_DB
+	if matches := useDatabaseQuery.FindStringSubmatch(query); len(matches) > 1 {
+		dbID := matches[1]
+
+		if err = c.UseDB(dbID); err == nil {
+			r = &my.Result{
+				Status:       0,
+				InsertId:     0,
+				AffectedRows: 0,
+				Resultset:    nil,
+			}
+		}
+
+		return
+	}
+
+	// as normal query
+	if readQuery.MatchString(query) {
+		var rows *sql.Rows
+		if rows, err = conn.Query(query); err != nil {
+			err = my.NewError(my.ER_UNKNOWN_ERROR, err.Error())
+			return
+		}
+
+		// build result set
+		return c.buildResultSet(rows)
+	}
+
+	var result sql.Result
+	if result, err = conn.Exec(query); err != nil {
+		err = my.NewError(my.ER_UNKNOWN_ERROR, err.Error())
+		return
+	}
+
+	lastInsertId, _ := result.LastInsertId()
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
var lastInsertId should be lastInsertID